### PR TITLE
Split the LICENSE and NOTICE as source-release version and binary-release version

### PR DESCRIPTION
**Pull request by jixuan1989**

### CODE REVIEW

**Comment by jixuan1989** [comment]
@JulianFeinauer  I think once our mentor confirm the correctness of the files, we need to cherry-pick the changes to release/0.8.0 and the master branch.

File: pom.xml [hunk 1]
```diff
@@ -217,6 +217,8 @@
                             <!-- if `mvn clean` is not executed and `mvn test` is interrupted, then perTestInputData may be not deleted -->
                             <exclude>src/test/resources/perTestInputData</exclude>
                             <exclude>src/test/resources/*.json</exclude>
+                            <exclude>NOTICE-binary</exclude>
+                            <exclude>LICENSE-binary</exclude>
```
Comment (qiaojialin) [inline pom.xml]:
```suggestion
                            <exclude>License-binary</exclude>
```

File: LICENSE [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+The following classes are modified from Apache Hive, which is under Apache License 2.0:
+
+iotdb/src/main/java/org/apache/iotdb/db/sql/parse/AstErrorNode.java
+iotdb/src/main/java/org/apache/iotdb/db/sql/parse/AstNode.java
+iotdb/src/main/java/org/apache/iotdb/db/sql/parse/AstNodeOrigin.java
+iotdb/src/main/java/org/apache/iotdb/db/sql/parse/Node.java
+iotdb/src/main/java/org/apache/iotdb/db/sql/parse/ParseDriver.java
+iotdb/src/main/java/org/apache/iotdb/db/sql/parse/ParseError.java
+iotdb/src/main/java/org/apache/iotdb/db/sql/parse/ParseException.java
+iotdb/src/main/java/org/apache/iotdb/db/sql/parse/ParseUtils.java
```
Comment (justinmclean) [inline LICENSE]: these paths are not correct for instance teh first file should be ./server/src/main/java/org/apache/iotdb/db/sql/parse/AstErrorNode.java

File: pom.xml (hunk 1 above)
Comment (jixuan1989) [reply pom.xml]: I think current file name is actually `LICENSE-binary`...

Comment (qiaojialin) [reply pom.xml]: this is because I changed that :)))

**Comment by jt2594838** [comment]
+1

**Comment by JulianFeinauer** [comment]
I would merge this now if there are no concerns left?

**Comment by JulianFeinauer** [comment]
+1

**Comment by jixuan1989** [comment]
has been cherry-picked to master by #309

**Pull request merged**